File: src/2956.find-common-elements-between-two-arrays.3206/solution.ts
function findIntersectionValues (nums1: number[], nums2: number[]): number[] {
    const s1 = new Set(nums1);
    const s2 = new Set(nums2);
    return [
        nums1.reduce((sum, item) => sum + (s2.has(item) ? 1 : 0), 0),
        nums2.reduce((sum, item) => sum + (s1.has(item) ? 1 : 0), 0),
    ];
}

File: src/2956.find-common-elements-between-two-arrays.3206/solution.java
class Solution {
    public int[] findIntersectionValues(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        for (int num : nums1) {
            set1.add(num);
        }
        Set<Integer> set2 = new HashSet<>();
        for (int num : nums2) {
            set2.add(num);
        }
        int a = 0;
        int b = 0;
        for (int n : nums1) {
            if (set2.contains(n)) {
                a++;
            }
        }
        for (int n : nums2) {
            if (set1.contains(n)) {
                b++;
            }
        }
        return new int[] { a, b };
    }
}